asf_installer_survey/notebooks/section_6_analysis.py

```diff
@@ -0,0 +1,1670 @@
+                                      df_p,
+                                     "SQ49",
+                                     ans_49
+                                     )
+df_615_crosstab_owner_V2 = df_615_crosstab_owner_V2.fillna(0)
+df_615_crosstab_owner_V2 = df_615_crosstab_owner_V2.astype(int)
+
+# Save to .csv and display
+df_615_crosstab_owner_V2.to_csv("../../outputs/section6/csv/615_owner_V2.csv")
+df_615_crosstab_owner_V2
+
+# %%
+# Generate figure
+groupedbar(df_615_crosstab_owner_V2,
+           "615_owner_V2",
+           "SQ 49a. Why do you intend to offer repairs, servicing and maintenance only for heat pump systems you have installed?"
+           )
+
+# %%
+# Generate figure
+donut(df_615_crosstab_owner_V2,
+           "615_owner_V2",
+           "SQ 49a. Why do you intend to offer repairs, servicing and maintenance only for heat pump systems you have installed?"
+           )
+
+# %%
+# CompanySizeEmployee V2
+# Create separate dataframe that stacks all answer columns
+df_p = pd.DataFrame()
+for n in range(0, len(ans_49)-1):
+
+    # Create new column for each answer option
+    response = ans_49[n]
+    new_col = "SQ49Ans"+str(n+1)
+    data[new_col] = data.apply(transform_49, axis=1)
+    
+    # Stack columns
+    df = data.filter(["CompanySizeEmployeeV2", new_col])
+    df = df.rename(columns={new_col:"SQ49"})
+    df_p = pd.concat([df_p, df])
+
+# Generate crosstab dataframe
+df_615_crosstab_employee_V2 = crosstable("CompanySizeEmployeeV2",
+                                         df_p,
+                                         "SQ49",
+                                         ans_49
+                                         )
+df_615_crosstab_employee_V2 = df_615_crosstab_employee_V2.fillna(0)
+df_615_crosstab_employee_V2 = df_615_crosstab_employee_V2.astype(int)
+
+# Save to .csv and display
+df_615_crosstab_employee_V2.to_csv("../../outputs/section6/csv/615_employee_V2.csv")
+df_615_crosstab_employee_V2
+
+# %%
+# Generate figure
+groupedbar(df_615_crosstab_employee_V2,
+           "615_employee_V2",
+           "SQ 49b. Why does the firm you work for intend to offer repairs, servicing and maintenance only for heat pump systems you have installed?")
+
+# %%
+# Generate figure
+donut(df_615_crosstab_employee_V2,
+           "615_employee_V2",
+           "SQ 49b. Why does the firm you work for intend to offer repairs, servicing and maintenance only for heat pump systems you have installed?")
+
+# %% [markdown]
+# ### RQ 6.16 | Why don’t those who only offer, or only intend to offer, repairs, servicing and maintenance to heat pump systems they’ve installed choose not to do so for other systems?
+# - SQ: 49a, 49b (intend to offer); 44a, 44b (offer)
+# - Sample:
+# - Standard_output
+# <p>
+# N.B. Merging two Select all that apply type crosstab dataframes
+# <p>
+# Update: Free text responses recoded for df_605_crosstab (SQ44a, 44b) on 2024-04-17
+
+# %%
+# Display dataframes to be merged
```

Comment:
Could provide here a brief motivation as to why we are merging these dataframes!

Reply:
Thanks Aidan, I've added some more clarity on the dataframes that are being merged. Hoping that the combination of the markdown explaining the research question section and the inline comments (which I know should be avoided but think they're most appropriate here) make it clearer :)

```
### RQ 6.24 | Why don’t survey participants offer repairs, servicing and maintenance (including those who are intending to do so in the next 12 months)?
- SQ: 50, 52a, 52b
- Merging the responses from two very similar questions: one which is asked to
those who intend (but don’t currently) offer repairs, servicing and
maintenance, and one for those who don’t intend to.
- Sample:
- Standard_output
<p>
N.B. Merging two Select all that apply type crosstab dataframes
<p>
Update: Free text responses recoded for df_618_crosstab on 2024-04-18
```
```
# Display dataframes to be merged
df_618_crosstab # don't currently offer but intending to
df_622_crosstab # don't curren't offer and don't intend to
```